pkg/container/startup.go

```diff
@@ -0,0 +1,68 @@
+package container
+
+import (
+	"fmt"
+	"strings"
+)
+
+type startup struct {
+	Entries map[string]entry `toml:"startup"`
+}
+
+type entry struct {
+	Name string
+	Args args
+}
+
+type args struct {
+	Name string
+	Dir  string
+	Env  map[string]string
+}
+
+func (e entry) Entrypoint() string {
+	if e.Name == "core.system" ||
+		e.Name == "core.base" && e.Args.Name != "" {
+		return e.Args.Name
+	}
+	return ""
+}
+
+func (e entry) WorkingDir() string {
+	return e.Args.Dir
+}
+
+func (e entry) Envs() []string {
+	envs := make([]string, 0, len(e.Args.Env))
+	for k, v := range e.Args.Env {
+		envs = append(envs, fmt.Sprintf("%s=%s", k, v))
+	}
+	return envs
+}
+
+// mergeEnvs merge a into b
+// all the key from a will endup in b
+// if a key is present in both, key from a are kept
+func mergeEnvs(a, b []string) []string {
+	ma := make(map[string]string, len(a))
```

Comment:
Why this has to be 2 dicts instead of one, the merge can done in the 2 loops directly but now u need 3 loops instead. Also did u test what will happen if u have `["KEY=V1", "KEY=V2"]` passed directly to process Env, because i think libc will just do the dedubing for u. this can be easily tested with one exec.Command line of `env`

Reply:
> Why this has to be 2 dicts instead of one, the merge can done in the 2 loops directly but now u need 3 loops instead. 

Right... :-)

>  Also did u test what will happen if u have ["KEY=V1", "KEY=V2"] passed directly to process Env, because i think libc will just do the dedubing for u.

I don't want to rely on a side effect when I can be sure to do it myself here.